### ISSUE TITLE
Accessibility Testing for Ask VA Integration - Ask VA

### ISSUE THREAD

**Issue by t-michaud** [department-of-veterans-affairs/va.gov-team#103653]
### Product information

- [x] Team name, product name, and feature name have been added to the title of this issue.
- [x] Team label, product label and feature label (if applicable) have been added to this issue.

### Who completed the use of color and color contrast test?

@t-michaud

### Use of color and color contrast checks

- [x] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [x] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [x] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [x] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [x] Any charts, maps, infographics, and tables convey all information without only relying on color
- [x] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### How did color testing go?

No issues. We use components and colors from the design system. All body text is dark on either white or light gray backgrounds. All alerts are from the design system too. So no issues with color contrast with text with colored backgrounds.

We don’t use any charts, maps, infographics, nor tables.

Nothing refers to color—including when providing instructions.

### Who completed the axe scans?

@t-michaud @joehall-tw

### axe checks

- [x] Each page has been scanned using axe (results shared below)
- [x] axe is integrated into your end-to-end testing

### axe DevTools scan results

@joehall-tw and our engineering team do have an automation of testing. 

The form tested in [Ask VA](https://staging.va.gov/contact-us/ask-va/introduction) is a step-by-step based on conditionals. Any direct link to a page will re-direct you back to the first page.

Despite regular automated testing, a few issues did arise.

## Issue 01: The "Country" select menu

On the “Your contact information” page, the `select` menu came up with an error. 
- Structure isn’t correct with the label not being within the `<va-select>` component (as [[seen in Storybook](https://design.va.gov/storybook/?path=/docs/uswds-va-select--required)](https://design.va.gov/storybook/?path=/docs/uswds-va-select--required))
- It does not go to the [[“read-only” state](https://design.va.gov/storybook/?path=/docs/uswds-va-select--read-only)](https://design.va.gov/storybook/?path=/docs/uswds-va-select--read-only) when the submitter selects “I live on a U.S. military base outside the US”
- Engineering is aware of the issue and will be working to correct the issues

## Issue 02: HTML structure issues 
On the Review and Submit page, there are 8 different issues regarding the  `DL` elements:
- There are some accordions where the `DL` elements contain `DL` elements

```
<dl> <!-- this parent DL needs to be changed to a DIV -->
  <dl>
    <dt></dt>
    <dd></dd>
  </dl>
</dl>
```

- Some `DL` elements contain `DIV` elements incorrectly

![Image](https://github.com/user-attachments/assets/3fd46e7d-02cf-4612-b76f-64717304c22d)

- Engineering is aware of the issue and will be working to correct the issues

### How did axe testing go?

The testing went very smoothly because our engineering team runs automated testing regularly, making our testing that much easier.

### Who completed the content zoom and reflow test?

@t-michaud

### Content zoom and reflow checks

- [x] All page elements are readable and usable at 200% zoom
- [x] All page elements are readable and usable at 300% zoom
- [x] All page elements are readable and usable at 400% zoom

### How did content zoom and reflow testing go?

No issues. Our pages and components are from the VA design system and scale without any issues. The page was visible zooming to 400%

### Who completed the keyboard navigation test?

@t-michaud

### Keyboard navigation checks

- [x] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [ ] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [x] All elements under focus have a visible focus indicator
- [x] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

Overall almost all pages with interactions passed the keyboard checks. However, there were a couple if issues

## Radio button + Modal

On the page "Who is your question about?" there are three radio buttons with three options:
- Myself
- Someone else
- General question

When navigating from "Myself" to "Someone else," a modal appears informing the submitter they have to sign in to ask a question about "Someone else" (same happens if you navigate from "Someone else" to "Myself") rather than waiting until the "Continue" button is clicked.

- The issue has been noted and may not be able to be fixed until after launch **NOT SURE IF THIS IS OKAY** 

## Search interaction using "Enter"

When performing a “Search” there were issues when hitting “enter” 
- Page: **School information** - error alert displays that I have to select a school vs. running a search
- Page: **Health facility** - after typing in a city, a modal displayed asking if I wanted to “exit” the form. OR sometimes will automaticallylly return me to the homepage of ASK VA.
- Engineering has been made aware of both issues

## Focus
The following pages didn’t have initial focus on the Progress bar’s `H2` “Complete with form” heading on
- Page: Check your mailing address
- Page: Review and submit (focus is on Ask VA)

### Do you have any other results to share?

Did testing with screenreaders and most pages performed without any issues.

## Review and Submit

The "Review and submit" page is missing the H2 heading
- Engineering has been made aware of this issue

## Authenticated dashboard/inbox

When a submitter is signed in and has asked a question when signing in, their previous questions are stored on the authenticated homepage. The word "Status" of the question is sometimes read -- if I can navigate to it correctly. The developers know the structure of the tab element + the H3 needs to be revised.

Current structure

```
<h3 class="vads-u-margin-top--0 vads-u-margin-bottom--0">
  <dl class="vads-u-margin-bottom--1p5">
    <dt class="sr-only">Status</dt>
    <dd><span class="usa-label vads-u-font-weight--normal vads-u-font-family--sans">In progress</span></dd>
  </dl>
  <span class="vads-u-display--block vads-u-font-size--h4 vads-u-margin-top--1p5">Submitted on [date]</span>
</h3>
```

New structure

But the card `H3` should be something like the following (using `SPAN` elements)

```
<h3 class="vads-u-margin-top--0 vads-u-margin-bottom--0">
  <div class="vads-u-margin-bottom--1p5">
    <span class="sr-only">Status</span>
    <span class="usa-label vads-u-font-weight--normal vads-u-font-family--sans">In progress</span>
    <span class="vads-u-display--block vads-u-font-size--h4 vads-u-margin-top--1p5">Submitted on [date]</span>
  </div>
</h3>
```

The Design System team is also working to include a `SPAN` element with the `SR-ONLY` attribute in an update.
